app/src/wpm.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/event_manager.h>
+#include <zmk/events/wpm_state_changed.h>
+#include <zmk/events/keycode_state_changed.h>
+
+#include <zmk/wpm.h>
+
+#define WPM_UPDATE_INTERVAL_SECONDS 1
+#define WPM_RESET_INTERVAL_SECONDS 5
+
+// See https://en.wikipedia.org/wiki/Words_per_minute
+// "Since the length or duration of words is clearly variable, for the purpose of measurement of
+// text entry, the definition of each "word" is often standardized to be five characters or
+// keystrokes long in English"
+#define CHARS_PER_WORD 5.0
+
+static uint8_t wpm_state;
+static uint8_t wpm_update_counter;
+static uint32_t key_pressed_count;
+
+int zmk_wpm_get_state() { return wpm_state; }
+
+int wpm_event_listener(const struct zmk_event_header *eh) {
+
+    LOG_DBG("Got an event");
```

Comment:
you may wanna clean up the debugs a bit, and maybe make them a bit less 'chatty', see other debug statements throughout the project.

Reply:
Good call - I was planning to drop the log message commit and put in some better logs 👍